proof/refine/ARM/CSpace1_R.thy

```diff
@@ -512,8 +513,6 @@ lemma cap_table_at_gsCNodes:
   apply blast
   done
 
-declare resolve_address_bits'.simps[simp del]
```

Comment:
I think we want this one globally -- although best declared directly after the definition of `resolve_address_bits'`, not in a much later theory here. Any specific reason to move this into the lemma context?

Reply:
I don't remember if I had a specific reason for this beyond noticing it and not liking the random declare in the middle of some unrelated proofs.  If you think we want to keep it globally I'm happy to move it to the definition as part of this PR. It seems like it won't affect anything else.

Reply:
Staring at the definition again, it should actually be fine to do this locally. I can't see a situation where you would want it automatically unfolded, but the case I was trying to avoid (non-terminating `simp` wherever `resolve_address_bits'` occurs) doesn't actually apply, because the first thing on the rhs of it is a big case statement.

It'd be slightly cleaner to move it to the definition site in ASpec, but I'm Ok either way.